cmd/go-judge/rest_executor/cmd_handler_test.go

```diff
@@ -0,0 +1,222 @@
+package restexecutor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/criyle/go-judge/cmd/go-judge/model"
+	"github.com/criyle/go-judge/envexec"
+	"github.com/criyle/go-judge/worker"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"slices"
+	"testing"
+	"time"
+)
+
+// mockWorker is a mock implementation of the worker.Worker interface
+type mockWorker struct {
+	// The result to send back when Submit is called
+	Result worker.Result
+}
+
+func (m *mockWorker) Submit(_ context.Context, req *worker.Request) (<-chan worker.Response, <-chan struct{}) {
+	// Mock implementation
+	rtCh := make(chan worker.Response)
+	go func() {
+		rtCh <- worker.Response{
+			RequestID: req.RequestID,
+			Results:   []worker.Result{m.Result},
+		}
+	}()
+	return rtCh, nil
+}
+
+func (m *mockWorker) Cancel(_ context.Context, _ string) error {
+	return nil
+}
+
+func (m *mockWorker) Close() {
+}
+
+func (m *mockWorker) Start() {
+}
+
+func (m *mockWorker) Execute(_ context.Context, _ *worker.Request) <-chan worker.Response {
+	return nil
+}
+
+func (m *mockWorker) Stat() worker.Stat {
+	return worker.Stat{
+		Queue:   10,
+		Running: 2,
+	}
+}
+
+func (m *mockWorker) Shutdown() {
+}
```

Comment:
You can just embed an interface and override the part you want to test to avoid duplicates. For example,

```go
type mockWorker struct {
    Result worker.Result
    worker.Worker
}
```